components/encryption/src/io.rs

```diff
@@ -387,8 +387,9 @@ impl CrypterCore {
 
     fn reset_buffer(&mut self, size: usize) {
         // OCrypter require the output buffer to have block_size extra bytes, or it will panic.
-        self.buffer.reserve(size + self.block_size);
+        #[allow(clippy::uninit_vec)]
```

Comment:
I don't think it's a safe implementation. /cc @hunterlxt

Reply:
Agree... Strictly speaking, every slice must be initialized. The problem is that the interface provided by the openssl library accepts a `& mut [u8]`. Passing an uninitialized buffer is UB but should work correctly in real world...

Reply:
This is an output buffer, and uninit-ed vec is safe as long as it's not read before written to. Still, I'm not against using `resize` here, the perf difference should be negligible.

Reply:
Using `Vec::set_len` like this is not panic-safe and should be avoided. The data should be written *before* the `Vec::set_len` call. See https://github.com/rust-lang/rust-clippy/issues/4483

Reply:
I understand, but panic safety is only a blocker if we are storing value with non-trivial `Drop` implementation, which isn't the case here.